### DISCUSSION TITLE
Is Vetur alive?

### DISCUSSION THREAD

**Discussion by vadimcoder** [vuejs/vetur#3378] [General] [answered]
Hi everyone,
Vetur is the most popular extension for vue.js according to VSCode rating, but is it still relevant? The last commit was last year, no Vue3 support (?), no ```<script setup>``` support, etc... What do you guys think? I'm asking this question because I am considering moving to VS Code and is trying to find a solution that fits my needs fully.

**Comment by kingyue737** [comment]
It seems that vetur is dead. The officially recommended extension is [volar](https://vuejs.org/guide/typescript/overview.html#ide-support) now.

**Comment by tiger-githubb** [comment]
hello the Vetur family is not bad but in my opinion it would be necessary to innovate for that it is necessary in my humble opinion to be inspired by volar which is the extension officially recommended

**Comment by jipika** [comment]
Vetur isn't updated anymore? Volar's support for VUe2 is terrible

**Comment by vadimcoder** [comment, marked as answer]
Looks like Vue2: Volar/Vetur (not recommended), Vue3: Volar only. Thanks to @kingyue737.

**Reply by kingyue737** [reply]
In fact, Volar works both for Vue2 and Vue3. With [@volar-plugins/vetur](https://github.com/johnsoncodehk/volar-plugins/tree/master/packages/vetur), you can get features of Vetur in Volar. I use Volar in all my Vue 2 projects. From my perspective, not recommended to use Vetur in new or actively maintained Vue 2 projects. As I mentioned in my answer, Vetur is dead, and its huge number of bugs won't be fixed.

**Reply by jipika** [reply]
Does this plug-in need to be installed in the project? Can't it exist in vscode ?

**Reply by kingyue737** [reply]
> Does this plug-in need to be installed in the project? Can't it exist in vscode ?

Yes, you need to install it in the project. It's plug-in, not built-in.

**Reply by vadimcoder** [reply]
@kingyue737 thanks, I updated my answer.

**Reply by kingyue737** [reply]
Volar v1.0 is released. For anyone who is interested at it, check out this post for more details https://blog.vuejs.org/posts/volar-1.0.html

**Reply by jipika** [reply]
Does vetur's global component configuration take effect with this plug-in

**Comment by yoyo930021** [comment]
Please check https://github.com/vuejs/vetur/issues/3476
